ionoscloud/utils.go

```diff
@@ -28,25 +28,188 @@ func resourceResourceImport(_ context.Context, d *schema.ResourceData, _ interfa
+
+		if err != nil {
+			return nil, err
+		}
+
+		if firewallRules.Items != nil {
+			if len(*firewallRules.Items) > 0 {
+				if err := d.Set("firewallrule_id", *(*firewallRules.Items)[0].Id); err != nil {
+					return nil, err
+				}
+			}
+		}
+
+		if firewallId, ok := d.GetOk("firewallrule_id"); ok {
+			firewall, _, err := client.NicApi.DatacentersServersNicsFirewallrulesFindById(ctx, datacenterId, serverId, primaryNic, firewallId.(string)).Execute()
+			if err != nil {
+				return nil, err
+			}
+
+			fw := map[string]interface{}{}
+			/*
+				"protocol": *firewall.Properties.Protocol,
+				"name":     *firewall.Properties.Name,
+			*/
+			setPropWithNilCheck(fw, "protocol", firewall.Properties.Protocol)
+			setPropWithNilCheck(fw, "name", firewall.Properties.Name)
+			setPropWithNilCheck(fw, "source_mac", firewall.Properties.SourceMac)
+			setPropWithNilCheck(fw, "source_ip", firewall.Properties.SourceIp)
+			setPropWithNilCheck(fw, "target_ip", firewall.Properties.TargetIp)
+			setPropWithNilCheck(fw, "port_range_start", firewall.Properties.PortRangeStart)
+			setPropWithNilCheck(fw, "port_range_end", firewall.Properties.PortRangeEnd)
+			setPropWithNilCheck(fw, "icmp_type", firewall.Properties.IcmpType)
+			setPropWithNilCheck(fw, "icmp_code", firewall.Properties.IcmpCode)
+
+			network["firewall"] = []map[string]interface{}{fw}
+		}
+
+		networks := []map[string]interface{}{network}
+		if err := d.Set("nic", networks); err != nil {
+			return nil, err
+		}
+	}
+
+	if server.Properties.BootVolume != nil {
+		if server.Properties.BootVolume.Id != nil {
+			if err := d.Set("boot_volume", *server.Properties.BootVolume.Id); err != nil {
+				return nil, err
+			}
+		}
+		volumeObj, _, err := client.ServerApi.DatacentersServersVolumesFindById(ctx, datacenterId, serverId, *server.Properties.BootVolume.Id).Execute()
+		if err == nil {
+			volumeItem := map[string]interface{}{}
+
+			setPropWithNilCheck(volumeItem, "name", volumeObj.Properties.Name)
+			setPropWithNilCheck(volumeItem, "disk_type", volumeObj.Properties.Type)
+			setPropWithNilCheck(volumeItem, "size", volumeObj.Properties.Size)
+			setPropWithNilCheck(volumeItem, "licence_type", volumeObj.Properties.LicenceType)
+			setPropWithNilCheck(volumeItem, "bus", volumeObj.Properties.Bus)
+			setPropWithNilCheck(volumeItem, "availability_zone", volumeObj.Properties.AvailabilityZone)
+
+			volumesList := []map[string]interface{}{volumeItem}
+			if err := d.Set("volume", volumesList); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	bootVolume, ok := d.GetOk("boot_volume")
+	if ok && len(bootVolume.(string)) > 0 {
+		_, _, err = client.ServerApi.DatacentersServersVolumesFindById(ctx, datacenterId, d.Id(), bootVolume.(string)).Execute()
+		if err != nil {
+			if err := d.Set("volume", nil); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if server.Properties.BootCdrom != nil {
```

Comment:
nil test performed on `server.Properties.BootCdrom` but we're dereferencing `server.Properties.BootCdrom.Id`, please add a nil check on `Id` as well.